### ISSUE TITLE
[IOS-XE] Additional spaces at the end of some test filenames break cloning on windows systems

### ISSUE THREAD

**Issue by JanisFalkenhagenWork** [CiscoTestAutomation/genieparser#833]
Hey there,

i noticed that some files have invalid filenames. Three of them have an additional space at the end, while one of them has only a backslash as filename.

The files in question are:

    "genieparser/src/genie/libs/parser/iosxe/tests/ShowEnvironmentStack/cli/empty/empty_output_output.txt "
    "genieparser/src/genie/libs/parser/iosxe/tests/ShowPlatformSoftwareFedIgmpSnooping/cli/empty/empty_output_output.txt "
    "genieparser/src/genie/libs/parser/iosxe/tests/ShowPlatformSoftwareFedActiveIpv6MldSnoopingVlan/cli/empty/empty_output_output.txt "
    and a file only containing a backslash as its name:
     "genieparser/src/genie/libs/parser/iosxe/tests/ShowIpDhcpBindingActiveCount/cli/equal/\"

This breaks cloning on windows systems because spaces at the end of filenames are not allowed. Also, backslashes are not allowed as filenames for obvious reasons on windows.

**Comment by SohanTirpude** [comment]
Hello @JanisFalkenhagenWork, 

We do not support pyats installation on Windows machines which you can check here as well: https://pypi.org/project/pyats/

![fjnESZ53fH](https://github.com/CiscoTestAutomation/genieparser/assets/13250447/6d9d9b26-ab52-4c39-bb40-464bc0f28e26)

We support only Linux and Mac machines. Kindly use either one of them.

Thank you.

**Comment by JanisFalkenhagenWork** [comment]
Hi @SohanTirpude 

I know, tgat you do not support Windows installation. That's not my point here. But those filenames are still broken. Linux might be able to handle those, but they are still not conforming to the naming conventions, especially the '\' file.

**Comment by JanisFalkenhagenWork** [comment]
Or am i wrong in assuming that those filenames are not not broken?

**Comment by SohanTirpude** [comment]
Hello @JanisFalkenhagenWork,

Sorry for the delay. As I was checking the same issue with my team. Yes, you are  correct about these files. Even I am not sure how these got added into the repo. 

Kindly give me some more time to check if there is any more such files in the repo or not, also to have enough checks in place to avoid such issues in future.

Thank you very much for finding out such issue.

**Comment by JanisFalkenhagenWork** [comment]
Hey,

no worries. Take the time you need.
This isn't an issue that needs fixing asap imo. Especially, since there is a pretty easy workaround with simply renaming the files locally.

But thank you for working on that issue regardless.

Thank you :D
I hope you had a nice easter weekend.

**Comment by SohanTirpude** [comment]
Hello @JanisFalkenhagenWork,

Sorry for late reply. I also hope you had a nice easter weekend. 

Well, the update is that we have merged the fix for this issue. Basically, we checked in our repos for files with trailing whitespace and so far we were able to find only the ones you reported and we fixed the issue in the file names. 

Thank you very much for having the keen eye and reporting back. Kindly let me know if you see any other issues on similar line.

Meanwhile, these files will be updated in the upcoming pyATS release hence, I am closing this ticket for now. If you see these files or any other files, kindly let me know.

Thank you.
